pypowsybl/network/impl/pandapower_converter.py

```diff
@@ -204,6 +206,51 @@ def create_shunts(n: Network, n_pdp: pandapowerNet) -> None:
         }, index=shunt_id)
         n.create_shunt_compensators(shunt_df=shunt_df, linear_model_df=linear_model_df)
 
+def extract_tap_info(trafo_and_bus):
+    """
+    This function ensures that the pandapower model was correct wrt phase_shifter boolean
+    Furthermore, the tap_step_percent is calculated from tap_step_degree which is
+    then augmented to the tap_step_percent already present, this means powsybl can just use
+    tap_step_percent for the calculations and then calculates the n_tap value
+    """
+    # Setup logging configuration
+    logging.basicConfig(level=logging.ERROR, format='%(levelname)s: %(message)s')
+
+    # Create a copy of 'tap_step_percent' to modify
+    tap_step_percent_new = np.zeros_like(trafo_and_bus['tap_step_percent'])
+
+    # Check if 'tap_step_degree' is not NaN (available) for transformers
+    degree_mask = ~np.isnan(trafo_and_bus['tap_step_degree'])
+
+    # Identify phase shift transformers (tap_phase_shifter == True)
+    phase_shift_mask = trafo_and_bus['tap_phase_shifter']
+
+    # Log error if both tap_step_degree and tap_step_percent are parameterized for phase shift transformers
+    both_parameterized_mask = phase_shift_mask & degree_mask & ~np.isnan(trafo_and_bus['tap_step_percent'])
+
+    if np.any(both_parameterized_mask):
+        for idx in trafo_and_bus.index[both_parameterized_mask]:
+            logging.error(
+                f"Transformer at index {idx} has both 'tap_step_degree' and 'tap_step_percent' parameterized.")
```

Comment:
IIDM model allows creation of a ratio tap changer AND a phase tap changer, we don't have to choose between the 2.

Reply:
yes, this log only suggests the user that in pandapower it is not allowed to set tap_step_degree as tap_step_percent. The motivation here is to help the user to debug the cases ogf incorrect network modelling.